mentor-mind-app/src/components/settings/SettingsMain.tsx

```diff
@@ -1,170 +1,238 @@
-          alignItems: "center",
-          justifyContent: "space-between",
-          width: "80%",
-          //   marginTop: -5,
-        }}
-      >
-        <Box
-          sx={{
-            margin: "5px",
-            flexDirection: "column",
-            width: "100%",
-            display: "flex",
-            alignItems: "center",
-          }}
-        >
-          <Box
-            sx={{
-              width: "100%",
-              //   marginLeft: "10px",
-            }}
-          >
-            <Typography component="h4" variant="h4" sx={{ marginTop: "10px" }}>
-              Settings
-            </Typography>
-          </Box>
+    const userId = load("userId");
+    const userPassword = load("userPassword");
+    const userRole = load("userRole");
 
-          <Box sx={{ width: "100%" }}>
-            <StyledContainer>
-              <FormLabel
-                htmlFor="settings-first-name"
-                sx={{
-                  width: "170px",
-                  color: "var(--secondary-color)",
-                  fontSize: "120%",
-                }}
-              >
-                First Name
-              </FormLabel>
-              <CssTextField
-                margin="dense"
-                required
-                fullWidth
-                id="settings-first-name"
-                placeholder="First name"
-                name="first-name"
-                autoComplete="off"
-                size="small"
-              />
-            </StyledContainer>
-            <StyledContainer>
-              <FormLabel
-                htmlFor="settings-last-name"
+    const handleInputChange = (e: { target: { name: any; value: any; }; }) => {
+        const { name, value } = e.target;
+        setFormData({
+            ...formData,
+            [name]: value,
+        });
+
+        if (name === 'email') {
+            setErrors({
+                ...errors,
+                email: validateEmail(value) ? '' : 'Invalid email address',
+            });
+        }
+    };
+
+    const validateEmail = (email: string) => {
+        const re = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+        return re.test(String(email).toLowerCase());
+    };
+
+    const handleSettingsChange = (e: { preventDefault: () => void; }) => {
+        e.preventDefault();
+
+        if (!validateEmail(formData.email)) {
+            setErrors({
+                ...errors,
+                email: 'Invalid email address',
+            });
+            return;
+        }
+
+        axios.post('http://localhost:8080/api/account/save', {
+            id: userId,
+            email: formData.email,
+            firstName: formData.firstName,
+            lastName: formData.lastName,
+            role: userRole,
+            password: userPassword
+        }, { withCredentials: false })
+            .then((response) => {
+                console.log('Info updated successfully:', response.data);
+                window.location.href = "/courses";
```

Comment:
Note to me: I will try to add a [snackbar](https://mui.com/material-ui/react-snackbar/) here, so that the user knows the update happened